### PR TITLE
Migrate custom object sync

### PR DESCRIPTION
**Pull request by salander85**
#### Summary
JIRA: https://commercetools.atlassian.net/browse/DEVX-109

#### Issues
The platform API for Custom object provides one single endpoint for create and update requests only. This differs from other platform resource APIs. For java-sync it means, for both operations the same method in `BaseService` is called. As the API may return a ConcurrentModification error in case of concurrent updates it was necessary to override the `BaseService.executeCreateCommand` method to submit the error to the caller in order to trigger retry mechanism.

### CODE REVIEW

File: src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+
+    final CustomObjectDraft newCustomObjectDraft =
+        CustomObjectDraftBuilder.of()
+            .container("someContainer")
+            .key("someKey")
+            .value(JsonNodeFactory.instance.objectNode().put("json-field", "json-value"))
+            .build();
+
+    // test
+    CustomObjectSyncStatistics syncStatistics =
```
Comment (lojzatran) [inline src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java:183]:
```suggestion
    final CustomObjectSyncStatistics syncStatistics =
```

File: src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+                CustomObjectCompositeIdentifier.of(newCustomObjectDraft).toString(),
+                "Failed to fetch from CTP while retrying after concurrency modification."));
+
+    verify(customObjectService).fetchCustomObject(any(CustomObjectCompositeIdentifier.class));
+    verify(customObjectService).upsertCustomObject(any());
+    verify(customObjectService).fetchMatchingCustomObjects(any());
+  }
+
+  @Test
+  void sync_UpdateWithSphereExceptionAndRetryWithFetchException_ShouldIncrementFailed() {
```
Comment (lojzatran) [inline src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java:319]:
```suggestion
  void sync_UpdateWithBadRequestExceptionAndRetryWithFetchException_ShouldIncrementFailed() {
```

File: src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+
+    // test
+    CustomObjectSyncStatistics syncStatistics =
+        new CustomObjectSync(spyCustomObjectSyncOptions, customObjectService)
+            .sync(singletonList(null))
+            .toCompletableFuture()
+            .join();
+
+    // assertion
+    assertThat(exceptions).hasSize(1);
```
Comment (lojzatran) [inline src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java:561]: The exceptions indeed has 1 element, but this element is null. Please verify.

**Review by lojzatran — CHANGES_REQUESTED** [review]

File: src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java (hunk 3 above)
Comment (salander85) [reply src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java:561]: [8d23794](https://github.com/commercetools/commercetools-sync-java/pull/997/commits/8d23794300064b1bd7ef3af85823c10dca4561d7)

File: src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java (hunk 2 above)
Comment (salander85) [reply src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java]: [8d23794](https://github.com/commercetools/commercetools-sync-java/pull/997/commits/8d23794300064b1bd7ef3af85823c10dca4561d7)

File: src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java (hunk 1 above)
Comment (salander85) [reply src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java]: [8d23794](https://github.com/commercetools/commercetools-sync-java/pull/997/commits/8d23794300064b1bd7ef3af85823c10dca4561d7)

File: src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java (hunk 3 above)
Comment (lojzatran) [reply src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java:561]: So null exception is expected? It's better to not pass any exceptions at all.

Comment (salander85) [reply src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java:561]: Ah, now i got the point.... Checkin...

Comment (salander85) [reply src/test/java/com/commercetools/sync/sdk2/customobjects/CustomObjectSyncTest.java:562]: [855d2dc](https://github.com/commercetools/commercetools-sync-java/pull/997/commits/855d2dcee69fa05066d328d155138cccbd87b7da)

**Review by lojzatran — APPROVED** [review]

**Pull request merged**